polkaj-scale/src/main/java/io/emeraldpay/polkaj/scale/writer/UInt64Writer.java

```diff
@@ -0,0 +1,31 @@
+package io.emeraldpay.polkaj.scale.writer;
+
+import io.emeraldpay.polkaj.scale.ScaleWriter;
+import io.emeraldpay.polkaj.scale.ScaleCodecWriter;
+import io.emeraldpay.polkaj.scale.reader.UInt64Reader;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+public class UInt64Writer implements ScaleWriter<BigInteger> {
+    @Override
+    public void write(ScaleCodecWriter wrt, BigInteger value) throws IOException {
```

Comment:
I mentioned above moving the read method to the shared BigIntReader. If that makes sense perhaps do the same thing for the writers along with the write method?

Reply:
The writers should just use constructor injection for the only thing varying which is the number of bytes. Then subclasses can pass in the length that pertains to them.